src/main/java/org/openstreetmap/atlas/checks/validation/intersections/SelfIntersectingPolylineCheck.java

```diff
@@ -144,4 +170,36 @@ else if (object instanceof Area)
     {
         return FALLBACK_INSTRUCTIONS;
     }
+
+    /**
+     * Returns true if adjacent {@link Edge} has identical lat,lng sequences
+     *
+     * @param polyline
+     *            the {@link PolyLine} being examined
+     * @return {@code true} if the any set of adjacent edges have identical geometries
+     */
+    private boolean hasDuplicateEdges(final PolyLine polyline)
+    {
+        final List<Segment> segments = polyline.segments();
+        final List<Segment> duplicates = new ArrayList<>();
+
+        // Loop through Polyline Segments
+        for (int i = 0; i < segments.size(); i++)
+        {
+            final Segment segment = segments.get(i);
+            final int adjacentEdge = i + 2;
+
+            // Check if segment exists elsewhere in List
+            if (segments.indexOf(segment) != segments.lastIndexOf(segment))
```

Comment:
What is the advantage of checking `segments.indexOf(segment) != segments.lastIndexOf(segment)` here. If there is a duplicate segment, then `adjacentEdge < segments.size() && segment.equals(segments.get(adjacentEdge))` would be true anyways. If `adjacentEdge < segments.size() && segment.equals(segments.get(adjacentEdge))` is true, then `segments.indexOf(segment) != segments.lastIndexOf(segment)` must be true. So `adjacentEdge < segments.size() && segment.equals(segments.get(adjacentEdge))` should be sufficient for duplicate check.

Reply:
Yes, good find. `segments.indexOf(segment) != segments.lastIndexOf(segment)` returns true if the segment exists elsewhere. In that case, I should be able to:

```java
if (segments.indexOf(segment) != segments.lastIndexOf(segment))
{
    return true;
}
```